modules/core/04-channel/types/expected_keepers.go

```diff
@@ -18,6 +18,8 @@ type ClientKeeper interface {
 
 // ConnectionKeeper expected account IBC connection keeper
 type ConnectionKeeper interface {
+	CheckIsOpen(ctx sdk.Context, connectionID string) error
+	HasConnection(ctx sdk.Context, connectionID string) bool
```

Comment:
Is `HasConnection` being used?

Reply:
it's actually not anymore, this was being used before the CheckIsOpen was introduced (there was a HasConnection check as well as a connection.GetSate() == OPEN) so if we keep CheckIsOpen we can remove HasConnection here